open-sphere-base/xyz-tile/src/main/java/io/opensphere/xyztile/envoy/XYZTileEnvoy.java

```diff
@@ -173,6 +177,40 @@ else if (LOGGER.isDebugEnabled())
         }
     }
 
+    /**
+     * Tests the envoy connection by sending a ping (i.e., a single get request)
+     * to the category's server.
+     *
+     * @param category Contains the server url information.
+     * @return The result of the envoy ping
+     */
+    public boolean ping(DataModelCategory category)
+    {
+        boolean pingSuccess = false;
+
+        ZYXImageKey key = new ZYXImageKey(0, 0, 0,
+                new GeographicBoundingBox(LatLonAlt.createFromDegrees(0., 45.), LatLonAlt.createFromDegrees(45., 90.)));
+        String urlString = buildImageUrlString(category, key);
+        URL url = UrlUtilities.toURL(urlString);
+
+        HttpServer server = getToolbox().getServerProviderRegistry().getProvider(HttpServer.class).getServer(url);
+        ResponseValues response = new ResponseValues();
+        try (CancellableInputStream stream = server.sendGet(url, response))
```

Comment:
You might try sending an HTTP HEAD request instead here. That's basically a GET with no body, and will cut down on transmitted data making the whole thing faster. I have examples of these requests within the AutoProxy wizard. Specifically, look at com.bitsys.mist.config.autoproxy.controller.ProxyWizardController.java:466: 
`          provider.getServer(url).sendHead(url, response);`
Which you could use here:
`          server.sendHead(url, response);`